core/src/AspectCore.Core/Injector/ServiceTable.cs

```diff
@@ -55,7 +55,11 @@ internal bool Contains(Type serviceType)
                 {
                     case Type enumerable when enumerable == typeof(IEnumerable<>):
                     case Type manyEnumerable when manyEnumerable == typeof(IManyEnumerable<>):
-                        return ContainsLinked(serviceType.GetTypeInfo().GetGenericArguments()[0]) ? true : true;
+                        if (ContainsLinked(serviceType.GetTypeInfo().GetGenericArguments()[0]))
+                        {
+                            return true;
+                        }
+                        break;
```

Comment:
This is the change that makes two tests fail.
I'm wondering why `ContainsLinked` is invoked in the original code if `true` is going to be returned despite the result of that invocation.

Reply:
This could be a bug. Let me think about it...

Reply:
Did you manage to get anything out of this, @liuhaoyang ?
I'm happy to remove it from this PR so that it gets merged and open a new one to properly investigate it.